### PR TITLE
Label destructive 'Dismiss' buttons red

### PR DESCRIPTION
**Pull request by NotMyFault**
This PR intends to distinguish better between primary buttons' with positive and negative operations. Dismissing warnings silences them forever. Making the button red draws more attention to it, like buttons used in other views themed red for irreversible actions.
Additionally, this gets a rid of multiple primary buttons with different, non-primary, tasks for the same notification window.

### Testing done

**Before**:
Screenshot 2022-11-11 at 23 17 30

**After**:
Screenshot 2022-11-12 at 10 44 29

### Proposed changelog entries

- Label 'Dismiss' buttons red.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7364"></a>

### CODE REVIEW

**Comment by janfaracik** [comment]
I definitely like the separation between the buttons +1 
I do wonder though if the use of a primary button for dismissing a notification is too prominent and as a result might draw too much attention to itself? Perhaps it'd look better if it wasn't using the '--primary' modifier - any thoughts?

**Comment by NotMyFault** [comment]
> Perhaps it'd look better if it wasn't using the '--primary' modifier - any thoughts?

Good call, it does indeed, most notably by dropping the primary effect.

**Review by timja — COMMENTED** [review]
lgtm, would be good if the UI for this was more standardised by default rather than all this copy and paste (well out of scope of this though)

any downstream PRs need filing for plugins?

**Comment by NotMyFault** [comment]
> any downstream PRs need filing for plugins?

I'm currently not aware of plugins out of my mind, but if I spot some, I'll happily propose upgrade PRs, as always.

**Comment by basil** [comment]
https://github.com/search?ref=simplesearch&type=code&q=user%3Ajenkinsci+%22submit+value%3D%5C%22%24%7B%25Dismiss%7D%5C%22%22

Needs downstream PRs in `active-directory-plugin` and `role-strategy-plugin`

**Comment by basil** [comment]
Not a complete list either:

```
~/src/jenkinsci $ rg -l 'submit .* value=".*Dismiss' | sort -u
dimensionsscm-plugin/src/main/resources/hudson/plugins/dimensionsscm/MissingJarsAdministrativeMonitor/message.jelly
ec2-plugin/src/main/resources/hudson/plugins/ec2/ssh/verifiers/SshHostKeyVerificationAdministrativeMonitor/message.jelly
ec2-plugin/src/main/resources/hudson/plugins/ec2/win/SelfSignedCertificateAllowedMonitor/message.jelly
localization-support-plugin/src/main/resources/io/jenkins/plugins/localization/support/LocalizationMonitor/message.jelly
matrix-auth-plugin/src/main/resources/org/jenkinsci/plugins/matrixauth/AmbiguityMonitor/message.jelly
opentelemetry-plugin/src/main/resources/io/jenkins/plugins/opentelemetry/administrativemonitor/ElasticBackendDashboardNotEnabledAdministrativeMonitor/message.jelly
opentelemetry-plugin/src/main/resources/io/jenkins/plugins/opentelemetry/administrativemonitor/ElasticBackendKibanaBaseUrlNotSetAdministrativeMonitor/message.jelly
opentelemetry-plugin/src/main/resources/io/jenkins/plugins/opentelemetry/administrativemonitor/ObservabilityBackendCheckAdministrativeMonitor/message.jelly
pipeline-maven-plugin/jenkins-plugin/src/main/resources/org/jenkinsci/plugins/pipeline/maven/NonProductionGradeDatabaseWarningAdministrativeMonitor/message.jelly
```

**Comment by NotMyFault** [comment]
> https://github.com/search?ref=simplesearch&type=code&q=user%3Ajenkinsci+%22submit+value%3D%5C%22%24%7B%25Dismiss%7D%5C%22%22
> 
> Needs downstream PRs in `active-directory-plugin` and `role-strategy-plugin`

The only repository I can submit a PR to would be opentelemetry, all other ones have an ancient baseline from years ago, which do not support the concept of `jenkins-button`s at all. They rely on the default design for everything.

**Comment by timja** [comment]
> The only repository I can submit a PR to would be opentelemetry, all other ones have an ancient baseline from years ago, which do not support the concept of `jenkins-button`s at all. They rely on the default design for everything.

It's just adding these attributes isn't it?

> primary="false" clazz="jenkins-!-destructive-color" 

If they aren't supported on those baselines they'll just be ignored.
IDEs will flag a warning but it won't cause any actual harm

**Comment by NotMyFault** [comment]
> It's just adding these attributes isn't it?

True that

**Review by timja — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Comment by basil** [comment]
Causes [JENKINS-70128](https://issues.jenkins.io/browse/JENKINS-70128).

**Pull request merged**
